### PR TITLE
[CLI-827] Module didn't register Error when switching node versions

### PR DESCRIPTION
**Pull request by feons**
https://jira.appcelerator.org/browse/CLI-827

### CODE REVIEW

File: lib/run.js [hunk 1]
```diff
@@ -12,10 +12,23 @@ var debug = require('debug')('appc:run');
...
  */
 function run (installBin, additionalArgs, cb, dontexit) {
-	var spawn = require('child_process').spawn,
+	var chalk = require('chalk'),
+		spawn = require('child_process').spawn,
 		path = require('path'),
 		util = require('./util');
 
+	if (installBin) {
+		debug('run check packge node version');
```
Comment (FokkeZB) [inline lib/run.js]: "a" missing

File: lib/run.js [hunk 2]
```diff
@@ -36,26 +39,60 @@ function run (installBin, additionalArgs, cb, dontexit) {
...
-				child.kill(name);
-				process.exit();
+				callback();
 			}
+		}
+	], function(error, results) {
+		if (error) {
+			console.log(chalk.yellow('The rebuild was unsuccessful, please run the following command to re-build for the newer version:'));
+			console.log(chalk.green('appc use %s --force\n'), util.getActiveVersion());
+			process.exit(1);
```
Comment (pinnamur) [inline lib/run.js]: I would not end the process if we weren't able to re-build the modules. We should keep the same behavior as it is now - just warn and continue.

File: lib/run.js [hunk 3]
```diff
@@ -36,26 +39,60 @@ function run (installBin, additionalArgs, cb, dontexit) {
 
 	debug('run with env=%j, args=%o',env,args);
 
-	// create our child process which simply inherits this process stdio
-	var child = spawn(process.execPath,args,{env:env, stdio:'inherit'});
+	async.series([
```
Comment (pinnamur) [inline lib/run.js]: Why are we having async.series even though we have one task in the array ?

**Comment by feons** [comment]
@pinnamur, updated according to comments. Please verify.

File: lib/run.js (hunk 3 above)
Comment (jhaynie) [inline lib/run.js]: looks like 2 tasks to me

File: lib/run.js [hunk 4]
```diff
@@ -36,27 +39,59 @@ function run (installBin, additionalArgs, cb, dontexit) {
...
 
-	// propagate signals to child process
-	['SIGTERM', 'SIGUSR1', 'SIGUSR2', 'SIGINT', 'SIGHUP', 'SIGQUIT', 'SIGABRT', 'exit'].forEach(function (name) {
-		process.on(name, function () {
-			debug('signal received', name, 'sending to', child);
-			if (name === 'exit') {
-				child.kill();
-				process.exit(arguments[0]);
+				debug('exec: %s in dir %s', cmd, packageDir);
+				exec(cmd,{cwd:packageDir, maxBuffer: 500 * 1024}, function(err, stdout, stderr){
```
Comment (jhaynie) [inline lib/run.js]: preferable to use spawn here. much cleaner code and you don't have to worry about this buffer overflow here.

**Pull request merged**